rafx-api/Cargo.toml

```diff
@@ -43,11 +43,21 @@ dispatch = { version = "0.2", optional = true }
 foreign-types-shared = { version = "0.1", optional = true }
 block = { version = "0.1.6", optional = true }
 
+# gl
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+raw-gl-context = { git = "https://github.com/aclysma/raw-gl-context.git", branch = "rafx", optional = true }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+#js-sys = "0.3.49"
+wasm-bindgen = "0.2.72"
+web-sys = { version = "0.3.4", features = ["Document", "Element", "HtmlCanvasElement", "WebGlBuffer", "WebGlTexture", "WebGlRenderingContext", "WebGlProgram", "WebGlShader", "WebGlRenderbuffer", "Window", "WebGlFramebuffer"] }
+
 [features]
 default = []
 rafx-empty = []
 rafx-vulkan = ["ash", "vk-mem", "ash-window"]
 rafx-metal = ["metal_rs", "objc", "raw-window-metal", "cocoa-foundation", "dispatch", "foreign-types-shared", "block"]
+rafx-gles2 = ["raw-gl-context"]
```

Comment:
One thing that I would try to do is actually squash this down into separate commits focused on each "layer" of the stack before merging. I think that'll make it easier to tell in the future what the different changes look like when adding a new back-end. 

e.g. like the visibility work https://github.com/aclysma/rafx/pull/125/commits

Reply:
Breaking it up will leave the repo in a broken state. (For example, changing the schema for the shader package will make all examples using them crash unless they are rebuilt.)

Reply:
> Breaking it up will leave the repo in a broken state. (For example, changing the schema for the shader package will make all examples using them crash unless they are rebuilt.)

It would only be broken in the _history_ on the PR branch. It wouldn't be broken at the PR branch's `HEAD` commit because everything is consistent by the last commit (so the PR build jobs would still pass), and it wouldn't be broken on any commits in master because you'll squash all of the PR commits during the merge leaving only the `HEAD` (so `git bisect` would still work), 

I don't feel especially strongly about it though.

Reply:
PR'ed the same thing in groups of files here: https://github.com/aclysma/rafx/pull/131 for historical purposes, but we will merge this one since the review comments are here and I trust this branch is correct more than that one that I just quickly threw together